src/chart/cartesianChart.js

```diff
@@ -0,0 +1,233 @@
+                var marginRight = yAxis.orient() === 'left' ? 0 : totalYAxisWidth;
+                svg.select('.x-axis')
+                    .layout({'height': xAxisHeight + xAxisLabelHeight, 'flexDirection': 'column',
+                        'marginLeft': marginLeft, 'marginRight': marginRight});
+
+                svg.select('.x-axis .label-container')
+                    .layout({'height': xAxisLabelHeight, 'alignItems': 'center', 'justifyContent': 'center'});
+
+                svg.select('.x-axis .axis')
+                    .layout('height', xAxisHeight);
+
+                // position the title
+                svg.select('.title')
+                    .layout({'height': chartLabelHeight, 'alignItems': 'center', 'justifyContent': 'center',
+                        'marginLeft': marginLeft, 'marginRight': marginRight});
+
+                // perform the flexbox / css layout
+                container.layout();
+
+                // update the label text
+                svg.select('.title .label')
+                    .text(chartLabel);
+
+                svg.select('.x-axis .label')
+                    .text(xAxisLabel);
+
+                svg.select('.y-axis.' + yAxis.orient() + ' .label')
+                    .text(yAxisLabel);
+
+                svg.select('.y-axis.' + opposingOrientation  + ' .label')
+                    .text('');
+
+                var plotAreaContainer = svg.select('.plot-area');
+
+                // configure the scales and render the axes
+                var xAxisContainer = container.select('.x-axis .axis');
+                xScale.range([0, plotAreaContainer.layout('width')]);
+                xAxisContainer.call(xAxis);
+
+                var yAxisContainer = svg.select('.y-axis.' + yAxis.orient() + ' .axis');
+                yScale.range([plotAreaContainer.layout('height'), 0]);
+                yAxisContainer.call(yAxis);
+
+                // remove the y axis from the opposing orientation
+                svg.select('.y-axis.' + opposingOrientation + ' .axis')
+                    .selectAll('*').remove();
+
+                // render the plot area
+                plotArea.xScale(xScale)
+                    .yScale(yScale);
+                plotAreaContainer.call(plotArea);
+
+                decorate(svg, data, index);
+            });
+        };
+
+        var scaleExclusions = [
+            'range', 'rangeRound', // the scale range is set via the component layout
+            'tickFormat'           // use axis.tickFormat instead
+        ];
+        fc.util.rebindAll(cartesianChart, xScale, 'x', scaleExclusions);
+        fc.util.rebindAll(cartesianChart, yScale, 'y', scaleExclusions);
+
+        // exclude scale because this component associates the scale with the axis, and
+        // exclude x axis orientation, as only y axis can be repositions to the left / right
+        fc.util.rebindAll(cartesianChart, xAxis, 'x', ['scale', 'orient']);
```

Comment:
Yes it would, although I think I've got a better idea for how to handle axis orientation, I'll just try out this alternative first.